### PR TITLE
TEP-0064 Tekton CLI workspace from local source

### PR DESCRIPTION
**Pull request by sm43**
This propose a TEP which proposes a way to use local source code in
tekton pipeline running in a kubernetes cluster.

POC [Use files on local machine in a pipeline (Experiment)](https://github.com/tektoncd/cli/pull/1334)

/cc @vdemeester 
 
Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

### CODE REVIEW

**Comment by afrittoli** [comment]
FYI @qu1queee

File: teps/0064-tekton-cli-workspace-from-local-source.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+### Use Cases 
+
+- As a developer, I have a Tekton Pipeline running in a kubernetes cluster. If there are some modification locally and I want to see how the CI behaves prior pushing the code to a repository or before creating a pull request.
+
+## Proposal
+
+One of the way, this can be achieved is
+- User provides the path to the source as a directory which has to be copied   
+- Create a volume. As pvc is most commonly used, we create a pvc with workspace name defined in the pipeline. The type and storage capacity of pvc can have default values and user can overide them if required.
```
Comment (pierretasci) [inline teps/0064-tekton-cli-workspace-from-local-source.md:50]: Doesn't this normally mean if my team has a Tekton CI job which normally does a `git clone` or similar as a first task, I need to rewrite my CI pipeline to skip that stage and take the PVC which already has data in it? That seems like it pushes more work onto the user.

File: teps/0064-tekton-cli-workspace-from-local-source.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- If the pvc by the workspace name already exists, user can provide a new name or override existing pvc
+- Spin up a long running pod and mount the pod on the pvc at `/workspace/<directory-name>`. The directory name would be available from the path to the directory provided.
+- Wait till the pod is in running state. 
+- Now, if pvc already existed and it needs to be override then use the `exec` command from `kubectl` pkg and execute `rm -rf /workspace/<directory-name>` inside the running pod. This will remove if their is an existing directoy in the pvc.
+- The code would be copied using the `cp` command from `kubectl` pkg where the source would the directory path provided and destination `/workspace/<directory-name>` inside the running pod.
+- Once the code is copied, the pod would be deleted.
+
+
+Image for Pod: The pod should be long running so a custom image can be created with `sleep infinty`.`tar` have to be installed as kubectl `cp` requires it.
```
Comment (pierretasci) [inline teps/0064-tekton-cli-workspace-from-local-source.md:59]: Many enterprises would not be ok with a custom image that they do not control or have mirrored locally to do this.

**Comment by vdemeester** [comment]
> @sm43
> If the first Task in a pipeline is `git-clone` (or something similar), does the git clone still happen and then get overridden by the copy, with this approach.

Yes it would, this is something we probably want to take into account somehow.

**Comment by pritidesai** [comment]
API WG discussion - @vdemeester - Lets experiment with CLI without changing API to evaluate the impact

**Comment by pritidesai** [comment]
@pritidesai create an issue and link to this PR, close this after we have an issue.

**Comment by pritidesai** [comment]
As per the discussion in the [API WG](https://docs.google.com/document/d/17PodAxG8hV351fBhSu7Y_OIPhGTVgj6OJ2lPphYYRpU/edit#heading=h.fjd0c3gb5yzo), creating an issue in CLI repo to track this PoC https://github.com/tektoncd/cli/issues/1399 and closing this PR.

/close

**Pull request closed without merging**